File: ova/10050/hartals.c
#include <stdio.h>
#include <stdlib.h>
#include <strings.h>
#include <stdbool.h>

int main(int argc, char **argv)
{
    int cases = 0, eff_hartals = 0;
    int days, parties, partie_h, i;
    bool daysh[3651];
    scanf("%d\n", &cases);


    while(cases > 0) {
        scanf("%d\n", &days);
        scanf("%d\n", &parties);
        bzero(daysh, 3651);
        
        while(parties > 0) {
        
            scanf("%d\n", &partie_h);
            for(i = 1; i <= days; i++) {
                if(i % 7 == 6 || i % 7 == 0) {
                    continue;
                }
                if(i % partie_h == 0 && !daysh[i]) {
                    daysh[i] = true;
                    eff_hartals++;
                }
            }

            parties--;
        }
        
        printf("%d\n", eff_hartals);
        cases--;
        eff_hartals = 0;
    }
    return 0;
}

File: ova/10050/hartals.py
import sys

if __name__ == "__main__":

    cases = int(input())

    for case in range(cases):
        days = int(input())
        parties = int(input())
        daysh = [False] * 3651
        hartals = 0

        for partie in range(parties):
            hartal = int(input())
            for day in range(1, days+1):
                if day % 7 == 6 or day % 7 == 0:
                    continue

                if day % hartal == 0 and not daysh[day]:
                    daysh[day] = True
                    hartals += 1

        print(hartals)

